src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type.equals(Bulldozer.class)) {
+            return (List<T>) new Bulldozer().get();
+        }
+        if (type.equals(Excavator.class)) {
+            return (List<T>) new Excavator().get();
+        }
+        if (type.equals(Truck.class)) {
+            return (List<T>) new Truck().get();
```

Comment:
use producers, not models
also, producers should be created as a class-level `private final` field